fireant/queries/builder/dataset_blender_query_builder.py

```diff
@@ -137,30 +138,43 @@ def _map_fields(fields):
 
             yield field.for_(field_map[field_from_blender])
 
-    dataset_metrics = _map_fields(metrics)
-
-    # Important: If no metrics are needed from this data set, then do not query
-    if not dataset_metrics:
-        return None
-
+    dataset_metrics = ordered_distinct_list_by_attr(_map_fields(metrics))
     dataset_dimensions = _map_fields(dimensions)
     dataset_filters = _map_fields(filters)
     dataset_references = _map_fields(references)
 
-    return (
-        dataset.query()
-        .widget(EmptyWidget(*dataset_metrics), mutate=True)
-        .dimension(*dataset_dimensions, mutate=True)
-        .filter(*dataset_filters, mutate=True)
-        .reference(*dataset_references, mutate=True)
+    if not any([dataset_metrics, dataset_dimensions]):
+        return [None]
+
+    # filter out operations that are relevant for this dataset
+    dataset_operations = []
+    dataset_metrics_aliases = [metric.alias for metric in dataset_metrics]
+    for operation in operations:
+        for metric in operation.metrics:
+            if metric.alias in dataset_metrics_aliases:
+                dataset_operations.append(operation)
+
+    share_dimensions = find_share_dimensions(dataset_dimensions, operations)
+
+    return make_slicer_query_with_totals_and_references(
+        dataset.database,
+        dataset.table,
+        dataset.joins,
+        dataset_dimensions,
+        dataset_metrics,
+        dataset_operations,
+        dataset_filters,
+        dataset_references,
+        [],
```

Comment:
may be worth specifying the keyword arg for this empty list just to make it clearer